db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -478,15 +481,6 @@ Status DBImpl::AtomicFlushMemTablesToOutputFiles(
   IOStatus log_io_s = IOStatus::OK();
   assert(num_cfs == static_cast<int>(jobs.size()));
 
-#ifndef ROCKSDB_LITE
-  for (int i = 0; i != num_cfs; ++i) {
-    const MutableCFOptions& mutable_cf_options = all_mutable_cf_options.at(i);
-    // may temporarily unlock and lock the mutex.
-    NotifyOnFlushBegin(cfds[i], &file_meta[i], mutable_cf_options,
```

Comment:
Can we keep the order of it? `SyncClosedLogs` involves IO and users use the hook to collect some stats such as flush duration or IO usage. For example, TiKV uses the hook to tag thread with specified IO type.

Reply:
That does a problem, how about moving `PickMemTable` here?

Reply:
That's what I mean. Though I didn't look through the code to check if the side effect of `PickMemTables` affects `SyncClosedLogs`.

Reply:
I found https://github.com/tikv/rocksdb/commit/c810947184679795a14a7f6bb6cb59a14c259cd2 mentioned this: 
```
Resolve the bug that if WAL sync fails, the memtable status does not roll back due to calling PickMemtable early than calling and checking SyncClosedLog.
```

Reply:
So it looks impossible to address the two issues at the same time. cc @tabokie

Reply:
I actually think it's better to move callback after sync logs (tagging it as IoType::Flush is not my expected behavior). Also I feel safer to move around callbacks than moving RocksDB code.